chemprop/args.py

```diff
@@ -123,6 +124,128 @@ class CommonArgs(Tap):
     """
     Path to constraints applied to atomic/bond properties prediction.
     """
+    is_pretrain: bool = False
+
+    """
+    Whether self supervised pretraining or not
+    """
+    not_skip_invalid_smiles: bool = False
+    """
+    Whether to filter smiles or not, if large data set it to False
+    """
+    is_pretrain_contra: bool = False
+    """
+    Whether self supervised pretraining use contrastive learning or not
+    """
+    is_pretrain_mask: bool = False
+    """
+    Whether self supervised pretraining use mask atom learning or not
+    """
+    mask_atom_pre_auto: bool = True
+    """
+    For self supervised pretraining whether the atom mask is done by auto
+    """
+    mask_bond_pre_auto: bool = True
+    """
+    For self supervised pretraining whether the bond mask is done by auto
+    """
+    MA_vocab_num_classes: int = 100
+    """
+    For self supervised pretraining the mask atom prediction has a number of possible classes ( Most basic is same as the max_atomic_number )
+    """
+    contrastive_output_size: int = 300
+    """
+    For self supervised pretraining the contrastive learning projection head output size. This dimension is used in calculating contrastive loss
+    """
+    Mode_of_contrastive_learning_probabilities: List = [1/6, 1/6, 1/6, 1/6, 1/6, 1/6]
+    """
+    For self supervised pretraining the contrastive learning different modes probabilities.
+    """
+    is_triplet: bool = False
+    """
+    For self supervised pretraining whether the triplet loss that has three views as input is added
+    """
+    triplet_margin: float = 0.0
+    """
+    For self supervised pretraining, the triplet loss margin parameter
+    """
+    triplet_p: int = 2
+    """
+    For self supervised pretraining, the triplet loss p parameter
+    """
+    triplet_loss_para: float = 0.1
+    """
+    For self supervised pretraining, the trade-off hyperparameter
+    """
+    MA_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the mask atom percentage for first view 
+    """
+    MA_second_percent: float = 0.3
+    """
+    For self supervised pretraining, the mask atom percentage for second view
+    """
+    BD_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the mask bond percentage for first view 
+    """
+    BD_second_percent: float = 0.3
+    """
+    For self supervised pretraining, the mask bond percentage for second view
+    """
+    SG_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the subgraph deletion percentage for first view 
+    """
+    SG_second_percent: float = 0.3
```

Comment:
Can we group these or some of these in the same way you group the probabilities above?